src/libs/SubscriptionUtils.ts

```diff
@@ -30,25 +119,220 @@ Onyx.connect({
     waitForCollectionCallback: true,
 });
 
-let ownerBillingGraceEndPeriod: OnyxEntry<number>;
-Onyx.connect({
-    key: ONYXKEYS.NVP_PRIVATE_OWNER_BILLING_GRACE_PERIOD_END,
-    callback: (value) => (ownerBillingGraceEndPeriod = value),
-});
-
-let amountOwed: OnyxEntry<number>;
-Onyx.connect({
-    key: ONYXKEYS.NVP_PRIVATE_AMOUNT_OWNED,
-    callback: (value) => (amountOwed = value),
-});
-
 let allPolicies: OnyxCollection<Policy>;
 Onyx.connect({
     key: ONYXKEYS.COLLECTION.POLICY,
     callback: (value) => (allPolicies = value),
     waitForCollectionCallback: true,
 });
 
+/**
+ * @returns The date when the grace period ends.
+ */
+function getOverdueGracePeriodDate(): OnyxEntry<number> {
+    return ownerBillingGraceEndPeriod;
+}
+
+/**
+ * @returns Whether the workspace owner has an overdue grace period.
+ */
+function hasOverdueGracePeriod(): boolean {
+    return !!ownerBillingGraceEndPeriod ?? false;
+}
+
+/**
+ * @returns Whether the workspace owner's grace period is overdue.
+ */
+function hasGracePeriodOverdue(): boolean {
+    return !!ownerBillingGraceEndPeriod && Date.now() < new Date(ownerBillingGraceEndPeriod).getTime();
+}
+
+/**
+ * @returns The amount owed by the workspace owner.
+ */
+function getAmountOwed(): number {
+    return amountOwed ?? 0;
+}
+
+function hasAmountOwed(): boolean {
+    return !!amountOwed;
```

Comment:
```suggestion
    return amountOwed !== undefined;
```

`!!amountOwed` will return `false` if `amountOwed` is `0`, is this intended? If not, is best to check if is different from `undefined`.

Reply:
My logic was that if `amountOwed` is 0 then user doesn't owe anything, so it should return false 😄